tests/dashboards/base_case.py

```diff
@@ -0,0 +1,114 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+from typing import Any, Dict, Union
+
+import prison
+from flask import Response
+
+from superset import app, security_manager
+from tests.base_tests import SupersetTestCase
+from tests.dashboards.consts import *
+from tests.dashboards.dashboard_test_utils import build_save_dash_parts
+from tests.dashboards.superset_factory_util import delete_all_inserted_objects
+
+
+class DashboardTestCase(SupersetTestCase):
```

Comment:
General comment: we should preferably not be introducing new class-based tests, but rather use functional `pytest` based tests. Check `tests/alerts_tests.py` for an example. Given that this is so far along in the making this is probably fine, but if it's possible to convert to proper `pytest` that would be awesome.

Reply:
I've looked into alerts_tests.py 
I understand its usefulles use by parameterizing the scenarios and I like it.
but, it seems to be using 
https://docs.pytest.org/en/stable/yieldfixture.html
which seems to be deprecated
is it still the right way to go?

Reply:
after deep diving in the differences between alerts and dashboard I noticed that alerts are being tests more like unit tests and dashboard security needs to be tests like integration tests .
to fully test we need the ability to tests against the http layer and to perform common actions like login and and other http calls which will need many abilities that comes out of the box in **SupersetTestCase** 
we will need to add test utils for functionall pytest in order support what currently exists. on class based test

Reply:
> I've looked into alerts_tests.py
> I understand its usefulles use by parameterizing the scenarios and I like it.
> but, it seems to be using
> https://docs.pytest.org/en/stable/yieldfixture.html
> which seems to be deprecated
> is it still the right way to go?

Oh, you're right, those are using the deprecated pattern. Should be possible to just replace `yield_fixture` with `fixture` (I can open a PR to clean those up).

Reply:
> after deep diving in the differences between alerts and dashboards tests I noticed that alerts are being tested more like unit tests and dashboard security needs to be tests like integration tests.
> to fully test we need the ability to tests against the http layer and to perform common actions like login and and other http calls which will need many abilities that comes out of the box in **SupersetTestCase**
> we will need to add test utils for functionall pytest in order support what currently exists. on class based test

Fair enough, and you may well be correct in that this will require a fair bit of new plumbing in the test utils before we can fully deprecate `SupersetTestCase`. I'm ok with keeping the current structure so we don't block this feature due to pending test pattern refactoring.